<?xml version="1.0"?>

<launch>
    <!-- Debug Info -->
    <arg name="debug" default="false" />
    <arg unless="$(arg debug)" name="launch_prefix" value="" />
    <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
    <arg name="robot" default="cob4-10"/>
    <arg name="robot_env" default="empty"/>
    <arg name="enable_gazebo_gui" default="false"/>
    <arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
    <arg name="component_name" default="arm_left"/>
    <arg name="debug_config_parameter" default="false"/>
    <arg name="debug_kinematic_cal" default="false"/>
    <arg name="debug_collision_detection" default="false"/>

    <group ns="$(arg component_name)">

        <!--node pkg="predictive_control" type="predictive_control_node" name="predictive_control_node" cwd="node" respawn="false" output="screen"/-->


        <group if="$(arg debug)">
                <node if="$(arg debug_config_parameter)" pkg="predictive_control" type="predictive_configuration_test" name="predictive_configuration_test" cwd="node" respawn="false" output="screen"/>
                <node if="$(arg debug_kinematic_cal)" pkg="predictive_control" type="kinematic_calculations_test" name="kinematic_calculations_test" cwd="node" respawn="false" output="screen"/>
                <node if="$(arg debug_collision_detection)" pkg="predictive_control" type="collision_detection_test" name="collision_detection_test" cwd="node" respawn="false" output="screen"/>
        </group>

    </group>
    <!-- configured rviz -->
    <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find predictive_control)/rviz_config/arm_left_desciption.rviz" /-->
</launch>
